.gemini/styleguide.md

```diff
@@ -0,0 +1,175 @@
+- **Don't repeat the PR description.** Only add new or clarifying information.
+- **Be brief unless needed.** Only write a longer summary if the PR description
+  is missing crucial details.
+- **Extend, don't duplicate.** If more detail is needed, clearly state what is
+  missing from the PR description and add only the necessary context.
+
+## Line Length and Formatting
+
+- **Maximum line length**: 120 characters (configured in ruff)
+- **Target Python version**: 3.10+
+- **Indentation**: 4 spaces per level
+
+## Type Hints
+
+- **Required**: All functions, methods, and class attributes must have type hints
+- **Avoid `Any`**: Use specific types whenever possible
+- **File operations**: Always parse file content into typed structures
+
+```python
+# Good - typed data structure
+@dataclass
+class Config:
+    model_name: str
+    temperature: float
+    max_tokens: int
+
+def load_config(path: Path) -> Config:
+    data = json.loads(path.read_text())
+    return Config(**data)
+
+# Bad - untyped dictionary
+def load_config(path: Path) -> dict[str, Any]:
+    return json.loads(path.read_text())
+```
+
+## Imports
+
+- **Group imports**: Standard library, third-party, local application
+- **Absolute imports**: Always use absolute imports for clarity
+- **Sort alphabetically** within groups
+- **Type checking imports**: Use `TYPE_CHECKING` for type-only imports
+
+```python
+import json
+import logging
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from requests import Response
+
+from cognite.client.data_classes import Asset
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+```
+
+## Naming Conventions
+
+- **Variables/functions**: `snake_case`
+- **Constants**: `UPPER_SNAKE_CASE`
+- **Classes**: `PascalCase`
+- **Modules**: `snake_case`
+- **Private members**: Single leading underscore `_private`
+
+## Docstrings
+
+Use concise docstrings with Args/Returns format. Based on repository patterns:
+
+```python
+def render_header(header: str) -> str:
+    """
+    Renders a (markdown) heading.
+
+    Args:
+        header (str): header
+
+    Returns:
+        str: The rendered header
+    """
+    return f"{header}\n{'=' * len(header)}\n"
+
+def walk_sdk_documentation(content: Tag, parser: Parser[T]) -> Iterable[T]:
+    """Parse the content of a file and yields documents. The parser controls how
+    the sections are transformed into documents."""
+    # Implementation here
+```
+
+**Docstring patterns**:
+
+- Start with a concise description
+- Use `Args:` and `Returns:` for complex functions
+- Omit obvious parameter descriptions
+- Keep descriptions brief and factual
+- Ok for `__init__` methods to omit docstring
+
+## Error Handling
+
+- **Specific exceptions**: Avoid broad `Exception` catches
+- **Graceful handling**: Provide meaningful error messages
+- **Type safety**: Return `None` or use Union types for fallible operations
+
+```python
+def load_llm_response(response: str) -> dict[str, Any] | None:
```

Comment:
I don't think this is a particularly good example. I would much rather have a descriptive exception raised than to have None returned and a warning logged.